### ISSUE TITLE
Wrap DOM event before passing it through to callbacks in KeyboardDragListeners

### ISSUE THREAD

**Issue by zepumph** [phetsims/scenery-phet#408]
This was found as I worked on https://github.com/phetsims/friction/issues/116#issuecomment-428163257.

I seem to remember in the past few months having a similar conversation about audio with @jonathanolson and @jbphet during a sonification meeting. Basically we should be wrapping the DOMEvent in something, just so that there is flexibility and futureproofing. 

Right now it looks like:
```js
      if ( self._start ) {
        if ( self.movementKeysDown ) {
          self._start( event );
        }
      }
```

and I think it would be fine to just do
```js
      if ( self._start ) {
        if ( self.movementKeysDown ) {
          self._start( {event} );
        }
      }
```
if the file is es6 of course :)

@jessegreenberg does this seem good to you? I am happy to implement this minor enhancement. Better to do it now before there are too many usages.

**Comment by jessegreenberg** [comment]
I'm sorry, I am sure this is reasonable, but I don't understand the recommendation. Is this object shorthand for { event: event }? How does this add flexibility?

**Comment by zepumph** [comment]
>Is this object shorthand for { event: event }

Yes that's the es6 way of saying that.

>How does this add flexibility?

In the same way that scenery "normal" event listeners pass a scenery `Event` type. `scenery/input/Event` takes in a domEvent arg, and wraps it, in effect "future proofing" any changes that may come from the listeners. If we decide that listeners need to be passed something else, we can just add it to our own, internal scenery Event type without changing the api. 

Right now for accessibility, since we just get the domEvent plain, we have no control over updates/changes to the api (since it is browser event specific). We also don't get the luxury of adapting the data that is passed to listeners without changing the arguments, such that we have to change all usages.

My recommendation is that we change all usages now, once, to something robust such that we would never have to do that sort of thing again.

Please let me know if this doesn't make sense, and we can talk about it more in person.

**Comment by jessegreenberg** [comment]
Thanks for the explanation, I think I understand now. 

> My recommendation is that we change all usages now, once, to something robust such that we would never have to do that sort of thing again.

In that case, should we go ahead and create an AccessibilityEvent (or something) in Scenery, rather than having a temporary wrap? Just trying to think through what this might look like at usages:
```js
someNode.addAccessibleInputListener( {
  keydown: ( event ) => {
    const a11yEvent = new AccessibilityEvent( event );

    // do stuff keycode specific stuff (maybe the keyCode would be assigned to our own AccessibilityEvent for convenience?)
    someFunction( a11yEvent.event.keyCode );

    // pass the event to an optional listener
    this.optionalListener && this.optionalListener( a11yEvent );
  }
} );
```
Or is there another way to create an AccessibilityEvent higher up in scenery?

**Comment by zepumph** [comment]
Yes good point! I think we have been down this road before. We decided that the ability to keep the same reference of the object added for the removal call was important, like:

```js
let myListener = {
  keydown: function( event ) {
    console.log( 'blarg!' );
  }
};

myNode.addAccessibleInputListener( myListener );
myNode.removeAccessibleInputListener( myListener );

```
vs
```js
let myWrappedListener = myNode.addAccessibleInputListener( myListener );
myNode.removeAccessibleInputListener( myWrappedListener );

```
which would allow us post processing manipulation, and the ability to wrap the event.


This issue was not trying to talk about that (though it is perhaps worth more discussion at some point). Instead I was talking about one level "higher", with the callbacks that are options passed to Typed listeners, like `KeyboardDragListener`: 
```js
function KeyboardDragListener( options ) {
    var self = this;

    options = _.extend( {

. . .

      // {Function|null} - Called as start( event: {DOMEvent} ) when keyboard drag is started
      start: null,

      // {Function|null} - Called as end( event: {DOMEvent}, viewDelta: {Vector2} ) when keyboard drag ends
      end: null, // called at the end of the dragging interaction

```

In this type, we have a keydown, but we could wrap that DOMEvent before giving it to these callbacks.

Yes what you mentioned above seems like it should be improved, and it is a code smell that of our listeners are interfacing directly with the DOMEvent, but in `KeyboardDragListener` here is a case that can be easily solved:
for example
```js

        // if movement keys are no longer down after keyup, call the optional end drag function
        if ( !moveKeysStillDown && moveKeysDown !== moveKeysStillDown ) {
          //self._end( event );
          self._end( new AccessibilityEvent( event ) );
        }
      }
```

What do you think?

**Comment by jessegreenberg** [comment]
My biggest concern is requiring AccessibilityEvent everywhere we use addAccessibleInputListener or create some accessibility input listener. While future proofing, I think it adds overhead without immediate benefit. That is why I was curious about wrapping the event somewhere in Scenery, not the listener.

**Comment by jessegreenberg** [comment]
Found this article, may be relevant: https://medium.com/@photokandy/til-you-can-pass-an-object-instead-of-a-function-to-addeventlistener-7838a3c4ec62

You can pass an object as the second argument of addEventListener as long as the object has a handleEvent method. So what if we changed the `addEventListener` call in `AccessibilityUtil.addDOMEventListeners` to 

```js
  domElement.addEventListener( event, new AccessibleInputHandler( accessibleInput[ event ] )  );
```

And AccessibleInputHandler looked like
```js

**Comment by zepumph** [comment]
I don't this the majority of this issue makes sense anymore. Now that a11y events are inlined with other scenery ones, it will be enough just to change the signature to `scenery/Event` instead of `DOMEvent`. @jessegreenberg does that sound good to you?

**Comment by jessegreenberg** [comment]
There is overlap between this issue and phetsims/tasks#970
